fullstop-plugins/fullstop-registry-plugin/src/main/java/org/zalando/stups/fullstop/plugin/RegistryPlugin.java

```diff
@@ -126,32 +133,69 @@ public void processEvent(final CloudTrailEvent event) {
                                 source, applicationVersionFromKio.getArtifact());
 
                         validateScmSource(event, applicationFromKio.getTeamId(), applicationId, applicationVersion);
+
+                        validateApprovals(applicationVersionFromKio, event);
                     }
                 }
             }
         }
     }
 
-    private void validateScmSource(CloudTrailEvent event, String teamId, String applicationId,
-            String applicationVersion) {
+    private void validateApprovals(Version version, CloudTrailEvent event) {
+        List<Approval> approvals = kioOperations.getApplicationApprovals(version.getApplicationId(), version.getId());
+
+        // #139
+        // https://github.com/zalando-stups/fullstop/issues/139
+        // does not have all default approval types
+        Set<String> approvalTypes = approvals.stream().collect(Collectors.groupingBy(Approval::getApprovalType))
+                .keySet();
+        if (!approvalTypes.contains("CODE_CHANGE") || !approvalTypes.contains("DEPLOY")
+                || !approvalTypes.contains("SPECIFICATION") || !approvalTypes.contains("TEST")) {
+
+            violationSink.put(new ViolationBuilder(format(
```

Comment:
You could probably use set operations here. E.g. remove all existing approvals from the set of required ones to build a more concise violation message.